### PR TITLE
Making whole navigation items clickable

### LINKED ISSUE #1154: You should be able to click anywhere inside nav item to activate link
## Issue Template

Right now, hovering over a link item will underline the text but you need to hover over the text itself in order to click the link.

## Expected

Clicking anywhere on the navigation link item will activate the link

### PR DESCRIPTION
**Pull request by bartaz**
## Done

- moved padding from `.p-navigation__link` to `<a>` element to make click area larger
- removed unnecessary font size change between breakpoints
- made sure to support navigation `<a>` links both nested in list elements and used directly with `.p-navigation__link` class

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- Go to navigation example (light or dark)
- Clicking on any part of navigation item should work (not just on link text)

## Details

Fixes #1154

## Screenshots

screen shot 2017-07-17 at 16 21 10

### CODE REVIEW

File: scss/_patterns_navigation.scss [hunk 1]
```diff
@@ -118,15 +118,9 @@
 
       & > a {
         // For when this class is applied to something that contains <a>s, e.g.:
-        // <li class="p-navigation__list"><a></a></li>
```
Comment (bartaz) [inline scss/_patterns_navigation.scss]: This class name was either typo or some leftover, and made reading code a bit confusing.

File: scss/_patterns_navigation.scss [hunk 2]
```diff
@@ -161,18 +154,28 @@
             border-bottom: 1px solid $color-mid-light;
           }
         }
-
-        @media (min-width: $breakpoint-navigation-threshold + 1) {
-          padding: $sp-small ($sp-large - $sp-xx-small);
-        }
       }
 
+      .p-navigation__link > a,
```
Comment (bartaz) [inline scss/_patterns_navigation.scss]: Applying same styles for links both nested under `.p-navigation__link` and directly under `.p-navigation__links`.

File: scss/_patterns_navigation.scss [hunk 3]
```diff
@@ -161,18 +154,28 @@
...
+        // </ul>
+        //
+        // and for links nested directly in navigation, e.g:
+        // <div class="p-navigation__links">
+        //   <a class="class="p-navigation__link"></a>
+        // </div>
+
+        color: $color-dark;
+        font-size: .875rem;
+        padding: .84375rem $sp-x-small;  // 0.84375rem = (48px [height] - 21px [line-height]) / 2
```
Comment (bartaz) [inline scss/_patterns_navigation.scss]: Custom padding of 13,5px (.84375rem) needs to be used to add up from 21px line-height to 48px desired height.

**Comment by bartaz** [comment]
Percy found visual diffs. But it's a new version that seems to be correct according to spec (uses 21px line-height instead of 24px in previous version).

File: scss/_patterns_navigation.scss (hunk 1 above)
Comment (barrymcgee) [inline scss/_patterns_navigation.scss]: Good spot 👌🏻

**Comment by barrymcgee** [comment]
LGTM - great work 👌🏻

**Pull request merged**